avm/res/recovery-services/vault/main.bicep

```diff
@@ -159,6 +189,23 @@ resource rsv 'Microsoft.RecoveryServices/vaults@2023-01-01' = {
     monitoringSettings: !empty(monitoringSettings) ? monitoringSettings : null
     securitySettings: !empty(securitySettings) ? securitySettings : null
     publicNetworkAccess: publicNetworkAccess
+    redundancySettings: !empty(redundancySettings) ? redundancySettings : null
+    restoreSettings: !empty(restoreSettings) ? restoreSettings : null
+    encryption: !empty(customerManagedKey)
+      ? {
+          infrastructureEncryption: 'Enabled'
+          kekIdentity: !empty(customerManagedKey.?userAssignedIdentityResourceId)
+            ? {
+                userAssignedIdentity: cMKUserAssignedIdentity.id
+              }
+            : {
+                useSystemAssignedIdentity: empty(customerManagedKey.?userAssignedIdentityResourceId)
+              }
+          keyVaultProperties: {
+            keyUri: cMKKeyVault.properties.vaultUri
```

Comment:
This seems to be an interesting case. @alexanderojala correct me if I'm wrong, but I read the [docs](https://learn.microsoft.com/en-us/azure/backup/encryption-at-rest-with-cmk?tabs=portal#:~:text=The%20update%20to%20the%20key%20version%20can%20take%20up%20to%20an%20hour%20to%20take%20effect) like the implementation can **only** do auto-key-rotation and you cannot actually specify a specific key version. If the case, we actually may need a different, simplified, interface. The current one. `customerManagedKeyWithAutoRotateType`, would seemingly allow a user to disable the auto-key-rotation, even though the code in the module actually cannot handle that case. 
@eriqua, thoughts?

If my assumption is correct, we'd just need another interface that would contain the key & key vault reference, as well as the managed identity...

Reply:
Not yet solved 😉